src/postgres/commands/configurePostgresFirewall.ts

```diff
@@ -7,43 +7,74 @@ import PostgreSQLManagementClient from "azure-arm-postgresql";
 import { FirewallRule } from "azure-arm-postgresql/lib/models";
 import * as publicIp from 'public-ip';
 import * as vscode from 'vscode';
-import { createAzureClient, DialogResponses, IActionContext } from "vscode-azureextensionui";
+import { createAzureClient, DialogResponses, IActionContext, IAzureQuickPickItem } from "vscode-azureextensionui";
 import { ext } from "../../extensionVariables";
+import { IPostgresWizardContext } from "../../tree/PostgresAccountWizard/IPostgresWizardContext";
 import { azureUtils } from "../../utils/azureUtils";
 import { localize } from "../../utils/localize";
 import { nonNullProp } from "../../utils/nonNull";
 import { PostgresServerTreeItem } from "../tree/PostgresServerTreeItem";
 
-export async function configurePostgresFirewall(context: IActionContext, treeItem?: PostgresServerTreeItem): Promise<void> {
-    if (!treeItem) {
+export async function configurePostgresFirewall(context: IActionContext | IPostgresWizardContext, treeItem?: PostgresServerTreeItem, createMode?: boolean): Promise<string> {
+    const ip: string = await publicIp.v4();
+    if (!treeItem && createMode) {
+
+        const yesPick: IAzureQuickPickItem<string> = { data: 'Yes', label: 'Yes' };
```

Comment:
I really hate yes/no quick picks - they just looks so ugly to me haha. For a more logical reason, I also prefer if the pick/button reflects the action users take rather than forcing them to parse out the question to figure out what "yes" does. How about this instead:

placeHolder: `A firewall rule is required to access this server from your current IP.`
pick1: `Add firewall rule for IP "{0}"`
pick2: `$(clock) Skip for now`

Reply:
What do you mean by "$(clock)"?

Reply:
It's an octicon: https://octicons.github.com/